gravitee-am-resource/gravitee-am-resource-smtp/src/main/java/io/gravitee/am/resource/smtp/provider/SmtpResourceProvider.java

```diff
@@ -96,16 +96,13 @@ private JavaMailSender createJavaMail() {
 
     @Override
     public Completable sendMessage(Email message, boolean overrideFrom) {
-        executorService.execute(() -> {
-            try {
-                if (overrideFrom) {
-                    message.setFrom(this.configuration.getFrom());
-                }
-                this.mailSender.send(message);
-            } catch (Exception e) {
-                LOGGER.error("Message emission fails", e);
-            }
-        });
-        return Completable.complete();
+        return Completable.fromFuture(
+                executorService.submit(() -> {
+                            if (overrideFrom) {
+                                message.setFrom(this.configuration.getFrom());
+                            }
+                            this.mailSender.send(message);
+                        }
+                ));
     }
```

Comment:
**todo**: Can you please have a look to all our EE plugin that implement the Resource interface and see if we have to update them please ?

Reply:
done